File: Code/BackEnd/Libs/Repositories/Models/TransferWorkflow.cs
using System;
using System.Collections.Generic;

namespace Repositories.Models;

public partial class TransferWorkflow : AuditableEntity
{
    public int Id { get; set; }

    public DateTime EffectiveAtUtc { get; set; }

    public DateTime BalanceDayUtc { get; set; }

    public int SourceAddressId { get; set; }

    public int DestinationAddressId { get; set; }

    public decimal SourceSurplusKwhAtWorkflow { get; set; }

    public decimal DestinationDeficitKwhAtWorkflow { get; set; }

    public decimal RemainingSourceSurplusKwhAfterWorkflow { get; set; }

    public decimal AmountKwh { get; set; }

    public int TriggerType { get; set; }

    public int Status { get; set; }

    public int SettlementMode { get; set; }

    public string? Notes { get; set; }

    public int AppliedDistributionMode { get; set; }

    public int? DestinationTransferRuleId { get; set; }

    public int? Priority { get; set; }

    public decimal? WeightPercent { get; set; }

    public virtual ICollection<TransferWorkflowStatusHistory> StatusHistory { get; set; } = new List<TransferWorkflowStatusHistory>();

    public virtual Address DestinationAddress { get; set; } = null!;

    public virtual Address SourceAddress { get; set; } = null!;

    public virtual DestinationTransferRule? DestinationTransferRule { get; set; }
}

File: Code/ReactUI/src/types/transferWorkflow.ts
export type TransferWorkflow = {
  id: number;
  effectiveAtUtc: string;
  balanceDayUtc: string;
  sourceAddressId: number;
  destinationAddressId: number;
  sourceSurplusKwhAtWorkflow: number;
  destinationDeficitKwhAtWorkflow: number;
  remainingSourceSurplusKwhAfterWorkflow: number;
  amountKwh: number;
  triggerType: number;
  status: number;
  notes: string | null;
  createdAtUtc: string;
  appliedDistributionMode: number;
  transferRuleId: number | null;
  priority: number | null;
  weightPercent: number | null;
};
